Source/stash.cpp

```diff
@@ -0,0 +1,618 @@
+			Stash.page += 10;
+			break;
+		}
+	}
+	if (Stash.page < 0)
+		Stash.page = 0;
+}
+
+void DoStashBtn()
+{
+	Rectangle stashButton;
+
+	for (int i = 0; i < 5; i++) {
+		stashButton = StashButtonRect[i];
+		stashButton.position = GetPanelPosition(UiPanels::Stash, stashButton.position);
+		if (stashButton.Contains(MousePosition)) {
+			StashButtons[i] = true;
+			IsStashButtonsPressed = true;
+		}
+	}
+}
+
+void DrawStash(const Surface &out)
+{
+	CelDrawTo(out, GetPanelPosition(UiPanels::Stash, { 0, 351 }), *StashPanelArt, 1);
+	for (unsigned buttonId = 0; buttonId < sizeof(StashButtons); buttonId++) {
+		if (StashButtons[buttonId]) {
+			Point stashButton = StashButtonCelRect[buttonId];
+			switch (buttonId) {
+			case 0:
+				CelDrawTo(out, stashButton, *StashNavButtonArt, 1);
+				break;
+			case 1:
+				CelDrawTo(out, stashButton, *StashNavButtonArt, 2);
+				break;
+			case 2:
+				CelDrawTo(out, stashButton, *StashGoldButtonArt, 1);
+				break;
+			case 3:
+				CelDrawTo(out, stashButton, *StashNavButtonArt, 3);
+				break;
+			case 4:
+				CelDrawTo(out, stashButton, *StashNavButtonArt, 4);
+				break;
+			}
+		}
+	}
+
+	for (int y = 0; y < 10; y++) {
+		for (int x = 0; x < 10; x++) {
+			if (Stash.stashGrids[Stash.page][x][y] != 0) {
+				InvDrawSlotBack(
+				    out,
+				    GetPanelPosition(UiPanels::Stash, StashCellPosition({ x, y })) + Displacement { 0, -1 },
+				    InventorySlotSizeInPixels);
+			}
+		}
+	}
+
+	for (int y = 0; y < 10; y++) {
+		for (int x = 0; x < 10; x++) {
+			if (Stash.stashGrids[Stash.page][x][y] > 0) { // first slot of an item
+				int ii = Stash.stashGrids[Stash.page][x][y] - 1;
+				int frame = Stash.stashList[ii]._iCurs + CURSOR_FIRSTITEM;
+
+				const auto &cel = GetInvItemSprite(frame);
+				const int celFrame = GetInvItemFrame(frame);
+				const Point position = GetPanelPosition(UiPanels::Stash, StashCellPosition({ x, y })) + Displacement { 0, -1 };
+				if (pcursstashitem == ii) {
+					CelBlitOutlineTo(
+					    out,
+					    GetOutlineColor(Stash.stashList[ii], true),
+					    position,
+					    cel, celFrame, false);
+				}
+
+				CelDrawItem(
+				    Stash.stashList[ii],
+				    out,
+				    position,
+				    cel, celFrame);
+			}
+		}
+	}
+
+	Point pos = GetPanelPosition(UiPanels::Stash, { 0, 0 });
+	for (auto &entry : StashEntries) {
+		if (entry.statDisplayFunc != nullptr) {
+			StyledText tmp = entry.statDisplayFunc();
+			DrawString(
+			    out,
+			    tmp.text,
+			    { entry.position + Displacement { pos.x, pos.y + 3 }, { entry.length, 18 } },
+			    UiFlags::AlignCenter | UiFlags::VerticalCenter | tmp.style, tmp.spacing);
+		}
+	}
+}
+
+void CheckStashItem(bool isShiftHeld, bool isCtrlHeld)
+{
+	if (pcurs >= CURSOR_FIRSTITEM) {
+		CheckStashPaste(MousePosition);
+	} else {
+		CheckStashCut(MousePosition, isShiftHeld, isCtrlHeld);
+	}
+	if (gbIsMultiplayer) {
+		sfile_write_stash();
+	}
+}
+
+int16_t CheckStashHLight(Point mousePosition)
+{
+	Point slot = { -1, -1 };

```

Comment:
`-1,-1` is used fairly frequently as a magic value, would be worth defining as a constant/constexpr value somewhere.

Reply:
I added `constexpr Point InvalidStashPoint { -1, -1 };` in stash.h